### PR TITLE
set correct change/seq on deduplicated nodes

### PR DESCRIPTION
**Pull request by noffle**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -353,7 +355,8 @@ Hyperlog.prototype.batch = function (docs, opts, cb) {
     })
 
     var added = nodes.length > 1 ? {} : null
-    var idx = 0
+    var seqIdx = 1
+    var changeIdx = 1
```
Comment (noffle) [inline index.js]: We need to increment these separately. When a new node is deduplicated locally, its seq++ but its change will be the same as the existing doc. However, when a brand new node is inserted (no dedupe), its seq AND change must increment.

File: index.js [hunk 2]
```diff
@@ -374,19 +377,21 @@ Hyperlog.prototype.batch = function (docs, opts, cb) {
         self.get(node.key, function (_, clone) {
           // It already exists, and it's in our local log.
           if (clone && clone.log === node.log) {
-            node.seq = clone.seq
+            node.seq = seq + (seqIdx++)
```
Comment (noffle) [inline index.js]: Deduped? Inc seq but use the old change value.

File: index.js [hunk 3]
```diff
@@ -372,21 +375,19 @@ Hyperlog.prototype.batch = function (docs, opts, cb) {
 
         // Check if the to-be-added node already exists in the hyperlog.
         self.get(node.key, function (_, clone) {
-          // It already exists, and it's in our local log.
-          if (clone && clone.log === node.log) {
-            node.seq = clone.seq
+          // It already exists
+          if (clone) {
+            node.seq = seq + (seqIdx++)
```
Comment (noffle) [inline index.js]: Already exists in the hyperlog? New seq#, but there's no new change to be inserted.

File: index.js [hunk 4]
```diff
@@ -372,21 +375,19 @@ Hyperlog.prototype.batch = function (docs, opts, cb) {
...
             node.change = added[node.key].change
             return fin()
-          // It's either brand new, or exists already in another log.
           } else {
-            node.seq = seq + 1 + idx
-            node.change = self.changes + 1 + idx
-            idx++
+            // new node across all logs
+            node.seq = seq + (seqIdx++)
+            node.change = self.changes + (changeIdx++)
```
Comment (noffle) [inline index.js]: Brand new node? Seq and change go up.

**Comment by mafintosh** [comment]
Looks great! Thanks for fixing all these bugs and adding more tests. I removed a console.log in the test.

4.10.5

**Comment by noffle** [comment]
> I removed a console.log in the test.

Ach! Every time. Thanks for catching that. :grinning:

**Pull request merged**